internal/resource/url.go

```diff
@@ -215,9 +215,12 @@ func (f *Fetcher) FetchFromTFTP(u url.URL, dest *os.File, opts FetchOptions) err
 // error if one is encountered.
 func (f *Fetcher) FetchFromHTTP(u url.URL, dest *os.File, opts FetchOptions) error {
 	if f.Client == nil {
-		f.Logger.Warning("Fetcher http client not initialized, ignoring any possible timeouts")
-		c := NewHttpClient(f.Logger, types.Timeouts{})
-		f.Client = &c
+		f.Logger.Warning("Fetcher http client not initialized, ignoring any possible timeouts and TLS")
+		if c, err := NewHttpClient(f.Logger, *f, types.Timeouts{}, nil); err != nil {
```

Comment:
What happens if `f.Client == nil` and the user has CA certs to fetch over http(s)? Would it trigger a call to `NewHTTPClient`, which would try to fetch the CAs via `f.Fetch`, which calls `f.FetchFromHTTP`, which would call `NewHTTPClient` because `f.Client` is still `nil`, which would call `f.Fetch`, and recurse indefinitely?

Reply:
Yeah......... I think the fetcher is going to need a little help. Let me think about this a bit more.